File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int MAX = Integer.MIN_VALUE, sum =0, n = nums.length;
        
        for(int i=0; i<n; i++) {
            sum += nums[i];
            MAX = Math.max(MAX, sum);
            
            if(sum<0) {
                sum = 0;
            }
        }
        
        return MAX;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int MAX = INT_MIN, sum = 0;
        int n = nums.size();
        // MAX = max(MAX, sum);
        
        for(int i = 0; i<n; i++) {
            sum = sum + nums[i];
            MAX = max(MAX, sum);
            
            if(sum < 0) {
                sum = 0;
            }
        }
        
        return MAX;
    }
};